dagshub/mlflow/patch.py

```diff
@@ -0,0 +1,176 @@
+import inspect
+import logging
+import sys
+import traceback
+from types import ModuleType, FunctionType
+from typing import Callable, Optional, Dict, Set
+
+from tenacity import Retrying, stop_after_delay
+
+logger = logging.getLogger(__name__)
+
+
+class MlflowMonkeyPatch:
+    context_manager_functions = ["start_run"]
+    classes_to_patch = ["mlflow.tracking.client.MlflowClient"]
```

Comment:
As we discussed, these lists should:
1. By default only include the O(n) unimportant logging functions - `log_artifact`, `log_artifacts`, `log_param`, `log_param`, `log_metric`, `log_figure`, `log_image`, `log_dict`, `log_table`, `log_text`
2. Be customizable (more about that below in the comment about `patch mlflow()`